src/components/button/index.tsx

```diff
@@ -1,27 +1,40 @@
-import React, { FC } from 'react';
+import React, { FC, MouseEventHandler } from 'react';
 import {
   Button as ChakraButton,
   ButtonProps as ChakraButtonProps,
 } from '@chakra-ui/react';
 
-export type ButtonProps = {
+type OnClick = MouseEventHandler<HTMLButtonElement>;
+
+type WithOnClick<T extends 'submit' | 'button'> = T extends 'submit'
```

Comment:
Optional: You could use the original definition instead of redefining the onClick with something like 
```
type WithOnClick<T extends 'submit' | 'button'> = Omit<ButtonProps, "onClick"> & T extends 'submit'
  ? { onClick?: ButtonProps['onClick'] }
  : { onClick: ButtonProps['onClick'] };
```

Reply:
No, because the `onClick` is optional on the `ButtonProps` so it would result in `MouseEventHandler | undefined`. I used the type that chakra uses in the re-definition

Reply:
Actually `Exclude` exists so I can exclude `undefined` from the union 🎉